src/navigation/courses/index.js

```diff
@@ -0,0 +1,217 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import {
+  Container,
+  Divider,
+  Item,
+  Header,
+  Button,
+  Segment,
+  Label,
+  Loader,
+  Dimmer,
+  Dropdown
+} from 'semantic-ui-react';
+import { Link } from 'react-router-dom';
+import { fetchCourses, deleteCourses } from './actions';
+import CourseLabels from './components/course-labels';
+import Filters from './components/filters';
+
+class Courses extends Component {
+
+  componentDidMount() {
+    this.props.deleteCourses();
+    let { course_type, page_token, category, category_id } = this.props.match.params;
+
+    // push to 404, if category doesn't match 'domain' or 'language'
+    if(course_type) {
+      if(course_type !== 'knowledge-base' && course_type !== 'soft-skills' && course_type !== 'random') {
+        this.props.history.push('/courses');
+      }
+    }
+    this.props.deleteCourses();
+    this.props.fetchCourses(course_type, page_token, category, category_id);
+  }
+
+
+  componentDidUpdate() {
+    let { course_type } = this.props.match.params;
+
+    // push to 404, if category doesn't match 'domain' or 'language'
+    if(course_type) {
+      if(course_type !== 'knowledge-base' && course_type !== 'soft-skills' && course_type !== 'random') {
+        this.props.history.push('/courses');
+      }
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    let { category, category_id, page_token, course_type } = nextProps.match.params;
+    if(this.props.match.params !== nextProps.match.params) {
+      // fetch data only if url parameters are different
+      this.props.deleteCourses();
+      this.props.fetchCourses(course_type, page_token, category, category_id);
+    }
+
+  }
+
+  filterCourses = (filters) => {
+    // filter courses using 'filters' props passed up by <Filters />
+    let { course_type, category, category_id } = this.props.match.params;
+    // let category_params = "";
+    // if(category && category_id) {
+    //   category_params = `/${category}/${category_id}`
+    // }
+    //
+    // //push to first page on filter
+    // this.props.history.push(`/courses/${course_type}/1${category_params}`);
+    this.props.deleteCourses();
+    this.props.fetchCourses(course_type, '1', category, category_id, filters);
+  }
+
+  renderSkillLevel(skill_level) {
+    if(skill_level === 'BG') {
+      return 'skill level: Beginner';
```

Comment:
Just like in reducer, try to avoid random strings in code. 

Instead, create constants in some file const BG = 'BG' and use them here. Always a good practice as you'll have to make changes at one places and using a const makes your intentions clear and code self documenting.